### PR TITLE
fix `tsh db connect` and `tsh proxy db` with logged in certs

### LINKED ISSUE #30486: tsh db connect - resource name collision should use best match
### Expected behavior:
The `tsh db connect` command should always resolve the database name prefix to the closest matching name.

### Current behavior:

When multiple databases share the same name prefix, as in the following scenario with `oracle*`, and one of them has an additional name suffix like `oraclevm1621` and  is active:
```bash
$ tsh  db connect --db-user=alice --db-name=TESTCDB oraclevm1621
$ tsh db ls
Name                                      Description                  Allowed Users Labels                                                                                                  Connect
----------------------------------------- ---------------------------- ------------- ------------------------------------------------------------------------------------------------------- ---------------------------
oracle                                                                 [*]           env=dev
oraclevm                                                               [*]           env=dev
> oraclevm1621 (user: alice, db: TESTCDB)                              [*]           env=dev                                                                                                 tsh db connect oraclevm1621
oraclevm2484                                                           [*]           env=dev

$ tsh db connect --db-user=alice --db-name=TESTCDB oraclevm
// BUG: connected to oraclevm1621 - db name was wrongly resolved due to active oraclevm1621
```
Executing the command: 
`tsh db connect --db-user=alice --db-name=TESTCDB oraclevm` 
will incorrectly resolve the `oraclevm` database name to `oraclevm1621`.

However, after logging out from oraclevm1621, the oraclevm database name gets resolved correctly:
```
tsh db logout oraclevm1621
tsh db connect --db-user=alice --db-name=TESTCDB oraclevm
// connected to oraclevm 
```

### PR DESCRIPTION
**Pull request by GavinFrazar**
fixes https://github.com/gravitational/teleport/issues/30486

functional change is just a single line of the diff, the rest is an update to tests.

When I originally made the changes that caused this issue, I actually intended for the behavior cited in the issue, but I can see how it would be confusing to a user.

### CODE REVIEW

**Comment by GavinFrazar** [comment]
flaky test detector failure is just a timeout issue. I've already done everything I can to speed up these tests and it's best that everything in `TestTshDB` runs in parallel for the overall test package speed. @smallinsky up to you, but I'd appreciate if you just comment `/excludeflake TestTshDB` :)

**Review by smallinsky — APPROVED** [review]

**Review by smallinsky — COMMENTED** [review]
> flaky test detector failure is just a timeout issue. I've already done everything I can to speed up these tests and it's best that everything in TestTshDB runs in parallel for the overall test package speed. @smallinsky up to you, but I'd appreciate if you just comment /excludeflake TestTshDB :)

@GavinFrazar 
I think that only GH admins can do that so we need to ask either @zmb3 or @r0mant

File: tool/tsh/common/db.go [hunk 1]
```diff
@@ -817,7 +817,7 @@ func onDatabaseConnect(cf *CLIConf) error {
 // is active in profile and no labels or predicate query are given.
 // Otherwise, the ListDatabases endpoint is called.
 func getDatabaseInfo(cf *CLIConf, tc *client.TeleportClient) (*databaseInfo, error) {
-	haveSelectors := len(tc.Labels) > 0 || tc.PredicateExpression != ""
+	haveSelectors := tc.DatabaseService != "" || len(tc.Labels) > 0 || tc.PredicateExpression != ""
```
Comment (greedy52) [inline tool/tsh/common/db.go:820]: isn't this always true?

**Review by greedy52 — APPROVED** [review]

File: tool/tsh/common/db.go (hunk 1 above)
Comment (GavinFrazar) [reply tool/tsh/common/db.go:820]: you can `tsh db login <db>` followed by `tsh db connect` without args, so this just preserves that behavior to select the active cert

**Pull request merged**